apps/chat-e2e/src/tests/chatExportImportWithAttachment.test.ts

```diff
@@ -502,3 +509,284 @@ dialTest(
+          conversationData.prepareConversationWithAttachmentsInRequest(
+            ModelIds.GPT_4_VISION_PREVIEW,
+            true,
+            gptProVisionImageUrl,
+          );
+        conversationData.resetData();
+        historyConversation = conversationData.prepareHistoryConversation(
+          geminiProVisionConversation,
+          gptProVisionConversation,
+        );
+        replayConversation =
+          conversationData.prepareDefaultReplayConversation(
+            historyConversation,
+          );
+        await dataInjector.createConversations([
+          historyConversation,
+          replayConversation,
+        ]);
+        await localStorageManager.setSelectedConversation(replayConversation);
+      },
+    );
+
+    await dialTest.step(
+      'Export replay conversation with attachments',
+      async () => {
+        await dialHomePage.openHomePage();
+        await dialHomePage.waitForPageLoaded();
+        await conversations.openEntityDropdownMenu(replayConversation.name);
+        await conversationDropdownMenu.selectMenuOption(MenuOptions.export);
+        exportedData = await dialHomePage.downloadData(() =>
+          conversationDropdownMenu.selectMenuOption(
+            MenuOptions.withAttachments,
+          ),
+        );
+      },
+    );
+
+    await dialTest.step(
+      'Remove all entities, import exported file and verify replay conversation is opened',
+      async () => {
+        await fileApiHelper.deleteAllFiles();
+        await chatBar.deleteAllEntities();
+        await confirmationDialog.confirm({ triggeredHttpMethod: 'DELETE' });
+        await dialHomePage.importFile(exportedData, () =>
+          chatBar.importButton.click(),
+        );
+        await conversationAssertion.assertEntityState(
+          { name: replayConversation.name },
+          'visible',
+        );
+        await recentEntitiesAssertion.assertReplayAsIsBordersColor(
+          Colors.controlsBackgroundAccent,
+        );
+        await chatAssertion.assertReplayButtonState('visible');
+      },
+    );
+
+    await dialTest.step(
+      'Replay conversation and verify attachments are sent in the requests',
+      async () => {
+        await dialHomePage.mockChatTextResponse(
+          MockedChatApiResponseBodies.simpleTextBody,
+        );
+        const replayRequests = await chat.startReplayForDifferentModels();
+        await apiAssertion.verifyRequestAttachments(
+          replayRequests[0],
+          geminiProVisionImageUrl,
+        );
+        await apiAssertion.verifyRequestAttachments(
+          replayRequests[1],
+          gptProVisionImageUrl,
+        );
+      },
+    );
+
+    await dialTest.step(
+      'Verify request attachments can be opened and downloaded',
+      async () => {
+        geminiProVisionAttachmentPath = `${API.importFileRootPath(BucketUtil.getBucket())}/${Attachment.sunImageName}`;
+        await chatMessages.expandChatMessageAttachment(
+          1,
```

Comment:
Do we consider hardcoded indexes a good approach?

Reply:
fixed